sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClient.cs

```diff
@@ -92,6 +92,72 @@ protected CommunicationIdentityClient()
             RestClient = null;
         }
 
+        /// <summary>Gets user details. <see cref="CommunicationUserIdentifier"/>.</summary>
+        /// <param name="communicationUser">Identifier of the identity. </param>"
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        public virtual Response<CommunicationIdentityResult> GetUser(CommunicationUserIdentifier communicationUser, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CommunicationIdentityClient)}.{nameof(GetUser)}");
+            scope.Start();
+            try
+            {
+                Response<CommunicationIdentityResult> response = RestClient.Get(communicationUser.Id, cancellationToken: cancellationToken);
+                var externalId = response.Value.ExternalId;
+                var responseId = response.Value.Id;
+                var lastTokenIssuedAt = response.Value.LastTokenIssuedAt;
+
+                return Response.FromValue(new CommunicationIdentityResult(responseId, externalId, lastTokenIssuedAt), response.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>Gets user details. <see cref="CommunicationUserIdentifier"/>.</summary>
+        /// <param name="communicationUser">The <see cref="CommunicationUserIdentifier"/> whose tokens should get revoked.</param>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        public virtual async Task<Response<CommunicationIdentityResult>> GetUserAsync(CommunicationUserIdentifier communicationUser, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CommunicationIdentityClient)}.{nameof(GetUserAsync)}");
+            scope.Start();
+            try
+            {
+                Response<CommunicationIdentityResult> response = await RestClient.GetAsync(communicationUser.Id, cancellationToken: cancellationToken).ConfigureAwait(false);
+                var externalId = response.Value.ExternalId;
```

Comment:
what about having the CommunicationIdentityResult construction in separate method, so we can call just the new method in the both sync and async public apis?